internal/collector/configmap.go

```diff
@@ -35,8 +35,8 @@ var (
 			Name: "kube_configmap_info",
 			Type: metric.MetricTypeGauge,
 			Help: "Information about configmap.",
-			GenerateFunc: wrapConfigMapFunc(func(c *v1.ConfigMap) metric.Family {
-				return metric.Family{
+			GenerateFunc: wrapConfigMapFunc(func(c *v1.ConfigMap) *metric.Family {
```

Comment:
Think we said at some point we also want to rename this `GenerateFunc` -> `Generator` or something along those lines?

Reply:
I think I'd generally be in favor of this, I think the common thing to do in go is have the `Genreator` be an interface and then have `GenerateFunc` be a typed function that implements the `Generator` interface.

Reply:
@LiliC yes, we talked about renaming `GenerateFunc` to `Generate` so that a `FamilyGenerator` `Generate`s families.

Instead I chose to introduce the `Generate` method on the `FamilyGenerator` (see comments):

```golang
// Generate calls the FamilyGenerator.GenerateFunc and gives the family its
// name. The reasoning behind injecting the name at such a late point in time is
// deduplication in the code, preventing typos made by developers as
// well as saving memory.
func (g *FamilyGenerator) Generate(obj interface{}) *Family {
	family := g.GenerateFunc(obj)
	family.Name = g.Name
	return family
}
```

By doing so we have the advantage of:

> Instead of making an external component responsible of adding the family name to
each family, enforce this in the generator via the Generate function.

Not sure we need the `Generator` interface abstraction at this point. If so, we could still do so at caller side, enforcing the signature of the above cited `Generate` function.

@LiliC @brancz what do you think?